EmailScraper.java

```diff
@@ -0,0 +1,163 @@
+ * The program will stop after a "reasonable" limit of pages have been parsed,                               
+ * which defaults to 100, but can be configured by commandline argument                                      
+ * (just pass in a single integer on the command line before the domain;                                     
+ *  we can use https://commons.apache.org/proper/commons-cli/ for real command                               
+ *  line parsing if we want to make this a real program). 
+ *
+ * @author George Ruban, george_ruban@yahoo.com
+ */
+public class EmailScraper {
+
+  /** The domain/website we're scraping emails off of. */
+  private String domain;
+
+  /** Domain in http format, for avoiding subdomains. */
+  private String httpDomain;
+
+  /** Domain in https format, for avoiding subdomains. */
+  private String httpsDomain;
+
+  /** Maximum number of pages to scrape. */
+  private int maxPages;
+
+  /** All gathered emails; a set, so no duplication. */
+  private Set<String> emails = new HashSet<>();
+
+  /** Pages we've already scraped. */
+  protected Set<String> visitedPages = new HashSet<>();
+  
+  /** Pages we've yet to scrape; a set, so no duplication. */
+  protected NavigableSet<String> futurePages = new TreeSet<>();
+
+  /**                                                                                                        
+   * @param domain Domain to scrape, for example www.jana.com                                                
+   * @param maxPages Maximum number of pages to scrape, -1 for infinite.                                     
+   */
+  public EmailScraper(String domain, int maxPages) {
+    this.maxPages = maxPages;
+    this.domain = domain.toLowerCase();
+    this.httpDomain = "http://" + this.domain;
+    this.httpsDomain = "https://" + this.domain;
+  }
+
+  /** Scrapes an entire domain for email addresses. */
+  public void scrape() {
+    futurePages.add(httpDomain);
+    while (futurePages.size() > 0 &&
+        (maxPages < 0 || visitedPages.size() < maxPages)) {
+      try {
+        scrapeUrl(futurePages.pollFirst());
+      } catch (IOException e) {
+        // Jana.com has this wonderful link: <a href="http:////www.jana.com/home"                            
+        // but no doubt other errors should also not crash us.                                               
+      }
+    }
+  }
+
+  /** Scrapes a specific URL for email addresses and links within the domain. */
+  private void scrapeUrl(String url) throws IOException {
+    visitedPages.add(url);
+    Document doc = Jsoup.connect(url).get();
+    addLinks(doc);
+    addEmails(doc);
+  }
+
+  /** 
+   * Queue up all the links that match the domain and                                                       
+   * haven't been traversed or queued yet.
+   */
+  protected void addLinks(Document doc) {
+    Elements links = doc.select("a[href]");
```

Comment:
When testing on http://janafin.fish/, the scraper actually picked up a couple emails that were not visible (hidden by a style of `display: None`):
```
contact+1@janafin.fish
contact@buckets.janafin.fish
```  	

Could you change this to not pick those emails up, since we are only looking for visible emails?